### ISSUE TITLE
[graphql] deploy fails where 'of: [{ type: 'stringType' }]' and stringType is of type string

### ISSUE THREAD

**Issue by relefant** [sanity-io/sanity#1478]
`sanity graphql deploy` fails with

`o: Bad Request - List definition for field "secondaryEmails" references unknown type "EmailAddress"
    at onResponse (~/.nvm/versions/node/v10.16.0/lib/node_modules/@sanity/cli/bin/sanity-cli.js:4522:533)`

When have the following items in my schema:

```js
export default {
  name: 'emailAddress',
  title: 'Email',
  type: 'string',
  validation: (Rule) => Rule.regex(/^[a-zA-Z0-9.!#$%&’*+/=?^_{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/),
};
```

```js
export default {
  name: 'business',
  title: 'Business',
  type: 'document',
  fields: [
    {
      name: 'name',
      title: 'Name',
      type: 'string',
    },
    {
      name: 'primaryEmail',
      title: 'Primary Email',
      type: 'emailAddress',
    },
    {
      name: 'secondaryEmails',
      title: 'Secondary Emails',
      type: 'array',
      of: [{ type: 'emailAddress' }],
    },
    {
      name: 'primaryTel',
      title: 'Primary Tel',
      type: 'string',
    },
    {
      name: 'secondaryTels',
      title: 'Secondary Tels',
      type: 'array',
      of: [{ type: 'string' }],
    },
    {
      name: 'primaryAddress',
      title: 'Primary Address',
      type: 'reference',
      to: [{ type: 'address' }],
    },
    {
      name: 'secondaryAddresses',
      title: 'Secondary Addresses',
      type: 'array',
      of: [
        {
          type: 'reference',
          to: [{ type: 'address' }],
        },
      ],
    },
    {
      name: 'openingHours',
      title: 'Opening Hours',
      type: 'openingHours',
    },
  ],
  preview: {
    select: {
      title: 'name',
      subtitle: 'primaryEmail',
    },
  },
};
```

If I change the type of the array for `secondaryEmails` from `emailAddress` to `string` then GraphQL API deploys successfully.

**Which versions of Sanity are you using?**
```
@sanity/cli                0.142.2 (up to date)
@sanity/base               0.142.2 (up to date)
@sanity/components         0.142.4 (up to date)
@sanity/core               0.142.4 (up to date)
@sanity/default-layout     0.142.4 (up to date)
@sanity/default-login      0.142.3 (up to date)
@sanity/desk-tool          0.142.4 (up to date)
@sanity/google-maps-input  0.142.4 (up to date)
```
**What operating system are you using?**

MacOS 10.14.6

**Which versions of Node.js / npm are you running?**

npm 6.10.1
node v10.16.0

**Comment by bjoerge** [comment]
Hi @relefant, thanks for reporting! We'll look into this.

**Comment by rexxars** [comment]
Fixed in v0.142.6
